pjlib/include/pj/stack.h

```diff
@@ -0,0 +1,189 @@
+/* 
+ * Copyright (C) 2008-2024 Teluu Inc. (http://www.teluu.com)
+ * Copyright (C) 2022-2024 Leonid Goltsblat <lgoltsblat@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA 
+ */
+#ifndef __PJ_STACK_H__
+#define __PJ_STACK_H__
+
+/**
+ * @file stack.h
+ * @brief Single Linked List data structure.
+ */
+
+#include <pj/types.h>
+
+#if defined(PJ_STACK_IMPLEMENTATION) && PJ_STACK_IMPLEMENTATION==PJ_STACK_WIN32
+
+/* the Windows platform stack implementation. */
+
+/* 
+ * PJ_STACK_ALIGN_PREFIX, PJ_STACK_ALIGN_SUFFIX is a readable syntax to declare
+ * platform default alignment for the stack item (see example below).
+ */
+
+#   define PJ_STACK_ALIGN_PREFIX PJ_ALIGN_DATA_PREFIX(PJ_POOL_ALIGNMENT)
+#   define PJ_STACK_ALIGN_SUFFIX PJ_ALIGN_DATA_SUFFIX(PJ_POOL_ALIGNMENT)
+
+#else  // PJ_STACK_IMPLEMENTATION != PJ_STACK_WIN32
+
+/* compilation of crossplatform stack implementation */
+
+#   define PJ_STACK_ALIGN_PREFIX
+#   define PJ_STACK_ALIGN_SUFFIX
+
+#endif // PJ_STACK_IMPLEMENTATION
+
+#include <pj/stack.h>
+
+
+PJ_BEGIN_DECL
+
+/*
+ * @defgroup PJ_DS Data Structure.
+ */
+
+/**
+ * @defgroup PJ_STACK Single Linked List with LIFO in/out policy
+ * @ingroup PJ_DS
+ * @{
+ *
+ * Stack in PJLIB is single-linked list with First In Last Out logic. 
+ * Stack is thread safe. Common PJLIB stack implementation uses internal locking mechanism so is thread-safe.
+ * Implementation for Windows platform uses locking free Windows embeded single linked list implementation.
+ * The performance of pj_stack implementation for Windows platform is 2-5x higher than cross-platform.
```

Comment:
Putting numbers here without test data may not be wise.
So may be just put "considerably" higher/faster.